Ubuntu/History/historymanager.cpp

```diff
@@ -0,0 +1,186 @@
+ */
+#include "historymanager.h"
+#include "manager.h"
+#include "eventview.h"
+#include <QDebug>
+
+
+HistoryManager::HistoryManager(QObject *parent) : QObject(parent), mFilter(0), mSort(new HistoryQmlSort(this)), mType(HistoryModel::EventTypeText), mDeletedCount(0), mPendingOperation(false), mSignalsTimer(-1)
+{
+}
+
+HistoryQmlFilter *HistoryManager::filter() const
+{
+    return mFilter;
+}
+
+void HistoryManager::setFilter(HistoryQmlFilter *value)
+{
+    if (mFilter) {
+        mFilter->disconnect(this);
+    }
+
+    mFilter = value;
+
+    Q_EMIT filterChanged();
+}
+
+HistoryModel::EventType HistoryManager::type() const
+{
+    return mType;
+}
+
+void HistoryManager::setType(HistoryModel::EventType value)
+{
+    mType = value;
+    Q_EMIT typeChanged();
+}
+
+HistoryQmlSort *HistoryManager::sort() const
+{
+    return mSort;
+}
+
+int HistoryManager::count() const
+{
+    return mCount;
+}
+
+int HistoryManager::deletedCount() const
+{
+    return mDeletedCount;
+}
+
+void HistoryManager::setSort(HistoryQmlSort *value)
+{
+    // disconnect the previous sort
+    if (mSort) {
+        mSort->disconnect(this);
+    }
+
+    mSort = value;
+
+    Q_EMIT sortChanged();
+}
+
+int HistoryManager::eventsCount()
+{
+    History::Filter queryFilter;
+
+    if (!mType || !mFilter) {
+        return false;
+    }
+
+    if (mFilter && mFilter->filter().isValid()) {
+        queryFilter = mFilter->filter();
+    }
+
+    return History::Manager::instance()->eventsCount((History::EventType)mType, queryFilter);
+}
+
+void HistoryManager::timerEvent(QTimerEvent *event)
+{
+    if (event->timerId() == mSignalsTimer) {
+        pendingOperation(false);
+    }
+}
+
+bool HistoryManager::removeAll()
+{
+
+    History::Filter queryFilter;
+    History::Sort querySort;
+
+    if (mPendingOperation) {
+        qWarning() << "there is a pending operation, request cancelled";
+        return false;
+    }
+
+    if (!mType || !mFilter || !mFilter->filter().isValid()) {
+        qWarning() << "type or filter not set, operation cancelled";
+        return false;
+    }
+
+    if (!mView.isNull()) {
+        mView->disconnect(this);
+    }
+
+    queryFilter = mFilter->filter();
+
+    if (mSort) {
+        querySort = mSort->sort();
+    }
+
+    setCount(eventsCount());
+    mDeletedCount = 0;
+
+    bool started = false;
+    if (count() > 0) {
+
+        mView = History::Manager::instance()->queryEvents((History::EventType)mType, querySort, queryFilter);
+        connect(mView.data(),
+                SIGNAL(eventsRemoved(History::Events)),
+                SLOT(onEventsRemoved(History::Events)));
+
+        started = History::Manager::instance()->removeEvents((History::EventType)mType, queryFilter, querySort);
+
+    }
+    pendingOperation(started);
+
+    return started;
+}
+
+void HistoryManager::onEventsRemoved(const History::Events &events)
+{
+    mDeletedCount += events.count();
+    qDebug() << "mDeletedCount" << mDeletedCount;
```

Comment:
This should be more descriptive, for example `"onEventsRemoved removed" << mDeletedCount << "so far";`. Otherwise it's unclear where the debug message is coming from and what it means.